### DISCUSSION TITLE
Feature Requests

### DISCUSSION THREAD

**Discussion by fvanroie** [HASwitchPlate/openHASP#35] [Ideas]
Here you can request new features and discuss ideas for hasp-lvgl.

**Comment by netmikey** [comment]
2 ideas immediately came to mind:

1. Add support for notifications. A notification would be some text that would wake up the plate and display the text prominently (like a modal dialog overlaid over the normal UI). Depending on its severity, a notification could stay on the screen until the user dismisses it, or it could auto-dismiss after some time.

2. Add a service for emitting sound using the Piezo Speaker on the ESP32 TouchDown. It would be nice to be able to specify the tone sequence and volume to be used. This could be used e.g. for getting the user's attention on important events. This could be combined with the notification idea above for some cool results.

**Reply by nagyrobi** [reply]
Msgbox for 1.?

**Comment by sleepypig2003** [comment]
Power-off detection and save state
--------------------------------------
It would be nice to be able to save state only if the power supply has been detected as being cut off.
i.e I'm thinking of a power supply off detector, maybe a signal on a GPIO?, then have a capacitor as backup power while openhasp writes its state to flash/storage somewhere and gracefully disconnects from homeassistant.

Anyone have any ideas how someone could implement something like this?

**Comment by hamster65** [comment]
I am considering switching from my own simple mqtt display-project to openHASP, but I am missing a pi chart widget to display energy usage and a graph over time-feature which supports 2-3 values, limit lines, scaling, bars, lines, filled etc.

For the pi chart, a simple function to draw circle segments would to. It would allow me to draw several segments with different start and stop angles, colors and radius to make some stick out.

**Reply by fvanroie** [reply]
We're using LVGL 7.11 in openHASP. The [Chart widget](https://docs.lvgl.io/7.11/widgets/chart.html) is not available yet through openHASP jsonl, but can be accessed using [custom code](https://www.openhasp.com/0.6.3/compiling/custom/).

For Pie Chart you need to use a [Canvas](https://forum.lvgl.io/t/how-to-use-lvgl-7-0-1-to-draw-a-sector/4271) in LVGL 7.11 or use [LVGL 8.x](https://docs.lvgl.io/8.3/widgets/extra/meter.html#pie-chart) which includes a pie chart widget.

**Reply by jwalker2020** [reply]
Is there any news on this? I would love an integrated Chart widget in openhasp.

**Comment by RestlessRabbits** [comment]
Would there be the possibility to have a list? A bit like the dropdown in an open state without the box?
I personaly would use this to have a scrollable list of Tasks and a shopping list on the display.
